feature/gnoi/packet_link_qualification/tests/packet_link_qualification_test/packet_link_qualification_test.go

```diff
@@ -256,36 +278,74 @@ func TestLinkQualification(t *testing.T) {
 	}
 	t.Logf("generatorCreateRequest: %v", generatorCreateRequest)
 
-	reflectorCreateRequest := &plqpb.CreateRequest{
-		Interfaces: []*plqpb.QualificationConfiguration{
-			{
-				Id:            plqID,
-				InterfaceName: dp2.Name(),
-				EndpointType: &plqpb.QualificationConfiguration_PmdLoopback{
-					PmdLoopback: &plqpb.PmdLoopbackConfiguration{},
-				},
-				Timing: &plqpb.QualificationConfiguration_Rpc{
-					Rpc: &plqpb.RPCSyncedTiming{
-						Duration: &durationpb.Duration{
-							Seconds: int64(plqDuration.testDuration.Seconds()),
-						},
-						PreSyncDuration: &durationpb.Duration{
-							Seconds: int64(plqDuration.preSyncDuration.Seconds()),
-						},
-						SetupDuration: &durationpb.Duration{
-							Seconds: int64(plqDuration.setupDuration.Seconds()),
-						},
-						PostSyncDuration: &durationpb.Duration{
-							Seconds: int64(plqDuration.postSyncDuration.Seconds()),
+	var reflectorCreateRequest *plqpb.CreateRequest
+	switch dut2.Vendor() {
+	case ondatra.JUNIPER:
```

Comment:
Can you factor out the commonality between the cases? It looks like all cases have a single interface in the request that differs only in the endpoint type? If so, much of that logic can be common.

Reply:
Updated per your suggestion. Thanks.